thirdeye/thirdeye-frontend/app/utils/pub-sub.js

```diff
@@ -0,0 +1,72 @@
+/** A very lightweight pub-sub implementation to communicate between ember-model-table
+ *  and the hosting component
+ *
+ *  Usage:
+ *  import pubSub from 'app/utils/pub-sub';
+ *
+ *  For publishing
+ *  pubSub.publish('testEvent', data);
+ *
+ *  For subscribing
+ *  const subscription = pubSub.subscribe('testEvent', (data) => {
+ *  });
+ *
+ *  For unsubscribing
+ *  subscription.unSubscribe();
+ */
+class PubSub {
```

Comment:
why you need a singleton to create an observer and subscriber here not an ember service and computed property to execute it?

Reply:
I had considered but did not do it because of 2 reasons:-
1. Because it is a temporary implementation until we have API ready to be able to do deep-linking. It is expected to go away.
2. Because if it turns out to be an implementation which lingers for a for longer due to unavailability of API in 1. above for a longer duration for any reason, then we don't want to do any additional initial refactoring as we transition to ember-octane. The more the services and computed properties, more tedious the refactoring effort.

Essentially, my idea is we want to keep new code lightweight and use as much vanilla javascript as we could to make our life easier as we transition towards ember octane. That means temporarily less reliance on native ember routines, services, computed properties and such.

Reply:
LGTM, but I am not sure this singleton pattern work, you could check whether every import shares the same instance.

Reply:
@zhangloo333 Every import shares the same instance. I have further validated it with a dry run.